python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2451,7 +2521,13 @@ def handle_new_event(
                 and event.get_dagster_event().is_job_event
             ):
                 self._run_storage.handle_run_event(run_id, event.get_dagster_event())
-
+                run = self.get_run_by_id(run_id)
+                if run and event.get_dagster_event().is_run_failure and self.run_retries_enabled:
```

Comment:
If we are going to use this tag to know whether the run is going to retry from e.g. airflift, don't we need the tag to be set even if run retries are not enabled?

Reply:
> I guess the tag just not being there at all also means that it is not going to retry

yeah that's right

probably worth documenting what the different values for the tag indicate. Where do you think is the right spot for that? `tags.py`?

Reply:
added a little comment where we define the tag, but can add some more info in other places if it seems relevant